### ISSUE TITLE
Nested TabNavigator Disappearing Randomly

### ISSUE THREAD

**Issue by crjackso** [react-community/react-navigation#1471]
**TL;DR**
I've got a nested tab navigator that shows up for a split second, then disappears for reasons I can't explain.

### Current Behavior
I have a navigation structure that looks something like so:
_AppNavigator.js:_
```javascript
export const AppNavigator = TabNavigator({
  welcome: { screen: WelcomeScreen },
  auth: { screen: AuthScreen },
  main: {
    screen: TabNavigator({
      deals: { screen: MyMap },
      myList: { screen: MyList }
    })
  }
}, getRootTabNavigatorOptions());

const AppWithNavigationState = ({ dispatch, nav }) => (
  <AppNavigator navigation={addNavigationHelpers({ dispatch, state: nav })} />
);

AppWithNavigationState.propTypes = {
  dispatch: PropTypes.func.isRequired,
  nav: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  nav: state.nav
});

export default connect(mapStateToProps)(AppWithNavigationState);
```
Inside my root js file, I render the overall app like so:
```javascript
  render() {
    return (
      <Provider store={this.store}>
        <AppWithNavigationState />
      </Provider>
    );
```

I have an action creator that dispatches a list of objects from Firebase to be rendered in the _MyList_ screen:
```javascript
export function dataFetch() {
  return (dispatch) => {
    getCurrentUser()
      .then((currentUser) => {
        const { uid } = currentUser;
        firebase.database().ref(`/users/${uid}/data`).on('value', (snapshot) => {
          dispatch({ type: DATA_FETCH_SUCCESS, payload: snapshot.val() });
        });
      });
  };
}
```
I've also got a nav reducer that looks like so:
```javascript
import { AppNavigator } from './../AppNavigator';

const initialState = AppNavigator.router.getStateForAction(AppNavigator.router.getActionForPathAndParams('welcome'));
  const nav = (state = initialState, action) => {
  const nextState = AppNavigator.router.getStateForAction(action, state);
  return nextState || state;
};

export default nav;
```
With this set up, I'm noticing the nested tab navigator flickers.  It shows up for a split second, then disappears.  I tried searching for a similar issue here, but couldn't find anything that seemed related.  One thing that is very interesting is, if I switch my action creator to something like what's listed below (without Firebase), the nested tab renders fine (and stays on the screen).  However, it's not immediately obvious to me why the action creator using Firebase causes an issue as it seems to be resolving the promise and returning data just fine.  As a result, I'm not convinced this is an issue I'm having with react-navigation per se, but I thought I'd post here to see if anyone has any suggestions or insight.

Thanks for your help!

_Temp action creator that works for some reason:_
```javascript
export const dataFetch = () => async (dispatch) => {
  try {
    const currentUser = await getCurrentUser();
    const { uid } = currentUser;

    const payload = {
      abc: { foo: 'bar' },
      def: { foo: 'baz' }
    };

    dispatch({ type: DATA_FETCH_SUCCESS, payload });
  } catch (e) {
    console.log('ERROR IN DATA FETCH!', e);
  }
};
```

### Expected Behavior

I expected the nested tab navigator to render correctly and not show momentarily before disappearing.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |  react-navigation@1.0.0-beta.9
| react-native     |  0.42.0
| node             | 7.5.0
| npm or yarn      | (npm) 4.5.0

**Comment by troute** [comment]
I have the same issue--to add to the strangeness, there is no problem when the tabBarPosition is set to 'top'. Any help would be greatly appreciated!

**Comment by XeniousSoftware** [comment]
Same issue here, looks like TabNavigator individual tab screen is overlapping on the bottom tabs.
can anyone help?

**Comment by XeniousSoftware** [comment]
i found the issue, the npm version of this repository is using react-native-tab-view 0.0.61 but the current repository is using react-native-tab-view 0.0.65. if you install directly from git then it fixes the issue.
i guess when they publish it next to npm then it should be fine.

**Comment by crjackso** [comment]
Yep, that seems to do it, @XeniousSoftware.   Thank you so much for the tip.  The nested tabs are no longer disappearing.  I updated my package.json to include the following for the time being until the npm repository is updated.

```
"react-navigation": "https://github.com/react-community/react-navigation.git"
```

**Comment by yaronlevi** [comment]
Also if you are using Yarn, do:
yarn add "https://github.com/react-community/react-navigation.git"

**Comment by bernatfortet** [comment]
I tried @crjackso and @yaronlevi and solutions but the same keeps happening to me.

**Comment by daivdggdev** [comment]
I have the same problem. I had install directly from git，but it‘s keep

**Comment by XeniousSoftware** [comment]
@bernatfortet @daivdggdev may be you can run `npm view react-native-tab-view` and see whats the version of react-native-tab-view. that may help you understand.

**Comment by bernatfortet** [comment]
I solved my issue.

What was happening is that I had 2 navigators using the same name for the specific screens.
Then, both navigators would listen for the same action (NAVIGATE) and execute navigation on their own reducer, resulting in 1 navigation action but 2 changes.

I'm not sure if it's the same issue as this one but the effect was similar.

I guess it make sense to use unique name for screens but it took me a while to realise that. I wish it was said in the documentation or prevented.
